lint/file_test.go

```diff
@@ -0,0 +1,225 @@
+package lint
+
+import (
+	"go/ast"
+	"go/token"
+	"testing"
+)
+
+func TestFile_disabledIntervals(t *testing.T) {
+	tests := []struct {
+		name     string
+		comments []*ast.CommentGroup
+		expected disabledIntervalsMap
+	}{
+		{
+			name: "no directives",
+			comments: []*ast.CommentGroup{
```

Comment:
nit: To simplify the definitions of use cases we could add a utility function that takes a `[]string` and returns `[]*ast.CommentGroup`.